yet-another-docker-its/src/main/java/com/github/kostyasha/it/SomeTest.java

```diff
@@ -0,0 +1,195 @@
+import com.github.kostyasha.yad.launcher.DockerComputerJNLPLauncher;
+import com.github.kostyasha.yad.strategy.DockerOnceRetentionStrategy;
+import hudson.cli.CLI;
+import hudson.model.FreeStyleProject;
+import hudson.model.Node;
+import hudson.remoting.Callable;
+import hudson.remoting.Channel;
+import hudson.slaves.Cloud;
+import hudson.slaves.JNLPLauncher;
+import hudson.tasks.Shell;
+import jenkins.model.Jenkins;
+import org.jenkinsci.remoting.RoleChecker;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.github.kostyasha.it.utils.JenkinsContainerUtil.createCliForInspect;
+
+/**
+ * @author Kanstantsin Shautsou
+ */
+public class SomeTest implements Serializable {
+    private static final long serialVersionUID = 1L;
+    private static final Logger LOG = LoggerFactory.getLogger(SomeTest.class);
+
+    Map<String, String> labels = new HashMap<String, String>() {{
+        put(SomeTest.class.getPackage().getName(), SomeTest.class.getName());
+    }};
+
+    String imageName = "jenkins:1.609.3";
+
+    @ClassRule
+    public static DockerRule d = new DockerRule(false);
+
+    @Test
+    public void dockerCloudDescriptorMissing() throws Exception {
+        // mark test method
+        CreateContainerCmd createCmd = d.cli.createContainerCmd(imageName)
+                .withPublishAllPorts(true)
+                .withLabels(labels);
+
+        d.pullImage(imageName);
+        final String id = d.createFreshJenkinsContainer(createCmd, true);
+
+        d.cli.startContainerCmd(id).exec();
+        final InspectContainerResponse inspect = d.cli.inspectContainerCmd(id).exec();
+        //TODO why CLI not Closeable?
+        final CLI cli = createCliForInspect(inspect, d);
+        cli.upgrade();
+
+        try (Channel channel = cli.getChannel()) {
+            LOG.info("Remoteclassloading allowed {}", channel.isRemoteClassLoadingAllowed());
+
+            LOG.info("Descriptor for Shell {}",
+                    channel.call(new CloudToStringCallable(){
+                        @Override
+                        public String call() throws Exception {
+                            final Shell shell = new Shell("");
+                            // works
+                            return shell.getDescriptor().toString();
+                        }
+                    }));
+
+            LOG.info("DockerCloud Descriptor: {}",
+                    channel.call(new CloudToStringCallable(){
+                        @Override
+                        public String call() throws Exception {
+                            final DockerCloud dockerCloud = new DockerCloud(null, null, 0, null);
+                            // fails
+                            return dockerCloud.getDescriptor().toString();
```

Comment:
@jglick could you help understand wtf is happening with classloaders for CLI? 
https://www.dropbox.com/s/yoycwil30w1cyy9/Screenshot%202015-12-13%2022.54.58.png?dl=0
Seems two classes/objects has different link because of different classloaders?
https://www.dropbox.com/s/krhzks6ihm2vxtg/Screenshot%202015-12-13%2023.11.21.png?dl=0
How i can manipulate with calls in the native master jenkins way?